cpp/src/parquet/arrow/schema_internal.h

```diff
@@ -39,8 +39,20 @@ Result<std::shared_ptr<::arrow::DataType>> GetArrowType(Type::type physical_type
                                                         const LogicalType& logical_type,
                                                         int type_length);
 
+// ARROW-12096 Exposing int96 arrow timestamp unit definition
+Result<std::shared_ptr<::arrow::DataType>> GetArrowType(Type::type physical_type,
+                                                        const LogicalType& logical_type,
+                                                        int type_length,
+                                                        const ::arrow::TimeUnit::type& int96_arrow_time_unit);
```

Comment:
I don't think this is the right place to pass int96-specific information. Perhaps this should be done at a higher level (for example in `schema.cc`?).

Reply:
Went back to to review this and not sure how to address.

Only thing I can imagine would be to add to `GetTypeForNode` (from `arrow/schema.cc`) and overwrite the standard `storage_type` if the parquet physical_type is INT96 and reader properties are not set to NANO? Let me know if I have misunderstood.